zk/stages/stage_sequence_execute.go

```diff
@@ -28,6 +29,78 @@ func SpawnSequencingStage(
 	cfg SequenceBlockCfg,
 	historyCfg stagedsync.HistoryCfg,
 	quiet bool,
+) (err error) {
+	roTx, err := cfg.db.BeginRo(ctx)
+	if err != nil {
+		return err
+	}
+	defer roTx.Rollback()
+
+	lastBatch, err := stages.GetStageProgress(roTx, stages.HighestSeenBatchNumber)
+	if err != nil {
+		return err
+	}
+
+	highestBatchInDS, err := cfg.datastreamServer.GetHighestBatchNumber()
+	if err != nil {
+		return err
+	}
+
+	if !cfg.zk.SequencerResequence || lastBatch >= highestBatchInDS {
+		if cfg.zk.SequencerResequence {
+			log.Info(fmt.Sprintf("[%s] Resequencing completed. Please restart sequencer without resequence flag.", s.LogPrefix()))
+			time.Sleep(10 * time.Second)
+			return nil
+		}
+
+		err = sequencingStageStep(s, u, ctx, cfg, historyCfg, quiet, nil)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Info(fmt.Sprintf("[%s] Last batch %d is lower than highest batch in datastream %d, resequencing...", s.LogPrefix(), lastBatch, highestBatchInDS))
+
+		batches, err := cfg.datastreamServer.ReadBatches(lastBatch+1, highestBatchInDS)
+		if err != nil {
+			return err
+		}
+
+		err = cfg.datastreamServer.UnwindToBatchStart(lastBatch + 1)
```

Comment:
In order to write to datastream only after everything is completed, a big refactoring seems required, because the datastream is currently updated in this function [updateStreamAndCheckRollback](https://github.com/0xPolygonHermez/cdk-erigon/blob/7ce4998778d9ba29ae9c0fe9a71f9414863b62ee/zk/stages/stage_sequence_execute.go#L518). 

As mentioned in the description of this PR, a solution to avoid losing old data stream is to backup the data stream files before resequence:
> Backup data stream files in case the files are corrupted during re-sequencing

Reply:
OK, considering this resequencing is really not going to be a common thing that makes sense that there are some manual steps.  It would also be recoverable from an RPC node so long as it was synced at the time.